windows/RNSVG/PatternView.cpp

```diff
@@ -55,77 +54,117 @@ void PatternView::UpdateProperties(IJSValueReader const &reader, bool forceUpdat
 
-    cmdList.Close();
+    com_ptr<ID2D1ImageBrush> imageBrush;
+    check_hresult(deviceContext->CreateImageBrush(cmdList.get(), brushProperties, imageBrush.put()));
+
+    auto transform{D2D1::Matrix3x2F::Translation({rect.left, rect.top})};
+    if (m_transformSet) {
+      transform = transform * D2DHelpers::AsD2DTransform(m_transform);
+    }
 
-    m_brush = brush;
+    imageBrush->SetTransform(transform);
+
+    copy_from_abi(m_brush, imageBrush.get());
   }
 }
 
-Windows::Foundation::Rect PatternView::GetAdjustedRect(Windows::Foundation::Rect const &bounds) {
-  float x{Utils::GetAbsoluteLength(m_x, bounds.Width) + bounds.X};
-  float y{Utils::GetAbsoluteLength(m_y, bounds.Height) + bounds.Y};
-  float width{Utils::GetAbsoluteLength(m_width, bounds.Width)};
-  float height{Utils::GetAbsoluteLength(m_height, bounds.Height)};
+D2D1_RECT_F PatternView::GetAdjustedRect(D2D1_RECT_F bounds) {
+  float width{bounds.right - bounds.left};
+  float height{bounds.bottom - bounds.top};
+
+  float x{Utils::GetAbsoluteLength(m_x, width) + bounds.left};
+  float y{Utils::GetAbsoluteLength(m_y, height) + bounds.top};
+  float adjWidth{Utils::GetAbsoluteLength(m_width, width)};
+  float adjHeight{Utils::GetAbsoluteLength(m_height, height)};
 
-  return {x, y, width, height};
+  return {x, y, adjWidth + x, adjHeight + y};
 }
 
-Microsoft::Graphics::Canvas::CanvasCommandList PatternView::GetCommandList(Windows::Foundation::Rect const &rect) {
-  auto const &root{SvgRoot()};
-  auto const &canvas{root.Canvas()};
-  auto const &cmdList{CanvasCommandList(canvas)};
-  auto const &session{cmdList.CreateDrawingSession()};
-  
-  if (m_align != "") {
-    Rect vbRect{
-        m_minX * root.SvgScale(),
-        m_minY * root.SvgScale(),
-        (m_vbWidth + m_minX) * root.SvgScale(),
-        (m_vbHeight + m_minY) * root.SvgScale()};
+com_ptr<ID2D1CommandList> PatternView::GetCommandList(ID2D1Device* device, D2D1_RECT_F rect) {
+  com_ptr<ID2D1DeviceContext> deviceContext;
+  check_hresult(device->CreateDeviceContext(D2D1_DEVICE_CONTEXT_OPTIONS_NONE, deviceContext.put()));
 
-    auto transform{Utils::GetViewBoxTransform(vbRect, rect, m_align, m_meetOrSlice)};
+  com_ptr<ID2D1CommandList> cmdList;
+  check_hresult(deviceContext->CreateCommandList(cmdList.put()));
 
-    if (m_transformSet) {
-      transform = transform * m_transform;
-    }
+  deviceContext->SetTarget(cmdList.get());
+
+  deviceContext->BeginDraw();
+  deviceContext->Clear(D2D1::ColorF(D2D1::ColorF::Orange, 0.0f));
 
-    session.Transform(transform);
+  auto transform{D2D1::Matrix3x2F::Identity()};
+
+  if (m_align != "") {
+    Rect vbRect{
+        m_minX,
+        m_minY,
+        (m_vbWidth + m_minX),
+        (m_vbHeight + m_minY)};
+
+    auto viewboxTransform{Utils::GetViewBoxTransform(
+        vbRect,
+        {
+            rect.left,
+            rect.top,
+            rect.left + rect.right,
+            rect.top + rect.bottom
```

Comment:
All the different rect types are making my head spin. Some LTRB, some XYWH. And here... why would you need left+rigth / top+bottom?